base/compiler/typeinfer.jl

```diff
@@ -442,15 +447,18 @@ function cache_result!(interp::AbstractInterpreter, result::InferenceResult)
             end
         end
     end
-    unlock_mi_inference(interp, mi)
     nothing
 end
 
 function cycle_fix_limited(@nospecialize(typ), sv::InferenceState)
     if typ isa LimitedAccuracy
         if sv.parent === nothing
-            # when part of a cycle, we might have unintentionally introduced a limit marker
-            @assert !isempty(sv.callers_in_cycle)
+            # we might have introduced a limit marker, but we should know it must be sv and other callers_in_cycle
+            #@assert !isempty(sv.callers_in_cycle)
+            #  FIXME: this assert fails, appearing to indicate there is a bug in filtering this list earlier.
+            #  In particular (during doctests for example), during inference of
+            #  show(Base.IOContext{Base.GenericIOBuffer{Memory{UInt8}}}, Base.Multimedia.MIME{:var"text/plain"}, LinearAlgebra.BunchKaufman{Float64, Array{Float64, 2}, Array{Int64, 1}})
+            #  we observed one of the ssavaluetypes here to be Core.Compiler.LimitedAccuracy(typ=Any, causes=Core.Compiler.IdSet(getproperty(LinearAlgebra.BunchKaufman{Float64, Array{Float64, 2}, Array{Int64, 1}}, Symbol)))
```

Comment:
AFAIU, this assertion should be failing because of the removal of `push!(frames, frame)` on L239, i.e. it fails when `LimitedAccuracy` is introduced in cases of self-recursion. Was this an intentional change?

Reply:
The code at L239 seemed to have been making this accidentally pass and hiding this bug